src/category_theory/comma.lean

```diff
@@ -229,6 +230,54 @@ variables {Y : T} {f : X ⟶ Y} {U V : over X} {g : U ⟶ V}
 @[simp] lemma map_map_left : ((map f).map g).left = g.left := rfl
 end
 
+section iterated_slice
+variables (f : over X)
+
+/-- Given f : Y ⟶ X, this is the obvious functor from (T/X)/f to T/Y -/
+@[simps]
+def iterated_slice_forward : over f ⥤ over f.left :=
+{ obj := λ α, over.mk α.hom.left,
+  map := λ α β κ, over.hom_mk κ.left.left (begin rw auto_param_eq, rw ← over.w κ, refl end)}
+
+/-- Given f : Y ⟶ X, this is the obvious functor from T/Y to (T/X)/f -/
+@[simps]
+def iterated_slice_backward : over f.left ⥤ over f :=
+{ obj := λ g, over.mk (over.hom_mk g.hom (by simp) : over.mk (g.hom ≫ f.hom) ⟶ f),
+  map := λ g h α, over.hom_mk (over.hom_mk α.left (over.w_assoc α f.hom))
+                              (over.over_morphism.ext (over.w α)) }
+
+/-- Given f : Y ⟶ X, we have an equivalence between (T/X)/f and T/Y -/
+def iterated_slice_equiv : over f ≌ over f.left :=
+equivalence.mk (iterated_slice_forward f) (iterated_slice_backward f)
```

Comment:
I wouldn't use `equivalence.mk` here. It potentially changes your unit and counit natural isos. But that shouldn't be needed here, and it's also not what you want. This equivalence is an adjoint equivalence on the nose.
So just use the
```lean
{ functor : ...,
  inverse : ... }
```
constructor notation. If the `functor_unit_iso_comp'` field gives you trouble, then you might need to write a couple of `simp`-lemmas for the two definitions above. (Or just battle through the proof.)

Reply:
When you do this, the two `@[simp]` lemmas immediately below can be replaced by adding `@[simps]` to this definition.